File: app/src/main/java/com/bogdanorzea/popularmovies/adapter/ReviewsAdapter.java
package com.bogdanorzea.popularmovies.adapter;

import android.content.Context;
import android.net.Uri;
import android.support.annotation.NonNull;
import android.support.customtabs.CustomTabsIntent;
import android.support.v7.widget.RecyclerView;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import android.widget.Toast;

import com.bogdanorzea.popularmovies.R;
import com.bogdanorzea.popularmovies.model.object.Review;

import java.util.ArrayList;
import java.util.List;

public class ReviewsAdapter extends RecyclerView.Adapter<ReviewsAdapter.ViewHolder> {

    private Context context;
    private List<Review> mReviews;

    public ReviewsAdapter(@NonNull Context context) {
        this.context = context;
        this.mReviews = new ArrayList<>();
    }

    public List<Review> getReviews() {
        return mReviews;
    }

    public void addReviews(@NonNull List<Review> data) {
        int size = getItemCount();
        mReviews.addAll(data);
        if (size > 0) {
            notifyItemRangeInserted(size, data.size());
        } else {
            notifyDataSetChanged();
        }
    }

    @NonNull
    @Override
    public ReviewsAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View layout = LayoutInflater.from(context).inflate(R.layout.item_review, parent, false);

        return new ReviewsAdapter.ViewHolder(layout);
    }

    @Override
    public int getItemCount() {
        return mReviews.size();
    }

    @Override
    public void onBindViewHolder(@NonNull ReviewsAdapter.ViewHolder holder, int position) {
        Review currentReview = mReviews.get(position);

        holder.itemView.setTag(currentReview);

        holder.author.setText(currentReview.author);
        holder.description.setText(currentReview.content);
    }

    static class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        final TextView author;
        final TextView description;

        public ViewHolder(View view) {
            super(view);

            author = view.findViewById(R.id.author);
            description = view.findViewById(R.id.content);

            view.setOnClickListener(this);
        }

        @Override
        public void onClick(View view) {
            Context context = view.getContext();

            Review review = (Review) itemView.getTag();

            if (!TextUtils.isEmpty(review.url)) {
                CustomTabsIntent.Builder builder = new CustomTabsIntent.Builder();
                CustomTabsIntent customTabsIntent = builder.build();
                customTabsIntent.launchUrl(context, Uri.parse(review.url));
            } else {
                Toast.makeText(context, "Couldn't find the review url", Toast.LENGTH_SHORT).show();
            }
        }

    }

}

File: app/src/main/java/com/bogdanorzea/popularmovies/ui/details/ReviewsAdapter.kt
package com.bogdanorzea.popularmovies.ui.details

import android.content.Context
import android.net.Uri
import android.support.customtabs.CustomTabsIntent
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import com.bogdanorzea.popularmovies.R
import com.bogdanorzea.popularmovies.model.`object`.Review
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.item_review.*

class ReviewsAdapter(private val context: Context) : RecyclerView.Adapter<ReviewsAdapter.ViewHolder>() {
    val reviews: MutableList<Review> = mutableListOf()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val layout = LayoutInflater.from(context).inflate(R.layout.item_review, parent, false)

        return ViewHolder(layout)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bindReview(reviews[position])
    }

    override fun getItemCount() = reviews.size

    fun addReviews(data: List<Review>) {
        val size = itemCount
        reviews.addAll(data)
        if (size > 0) {
            notifyItemRangeInserted(size, data.size)
        } else {
            notifyDataSetChanged()
        }
    }

    class ViewHolder(override val containerView: View) : RecyclerView.ViewHolder(containerView), View.OnClickListener, LayoutContainer {

        fun bindReview(review: Review) {
            author.text = review.author
            content.text = review.content

            containerView.tag = review
            containerView.setOnClickListener(this)
        }

        override fun onClick(view: View) {
            val context = view.context

            val (_, _, _, url) = itemView.tag as Review

            if (!url.isNullOrEmpty()) {
                CustomTabsIntent.Builder()
                        .build()
                        .launchUrl(context, Uri.parse(url))
            } else {
                Toast.makeText(context, "Couldn't find the review url", Toast.LENGTH_SHORT).show()
            }
        }

    }

}
